src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -1505,34 +1483,6 @@ private bool HasProblem(RegexNodeOrToken component)
             return false;
         }
 
-        private RegexExpressionNode ParseRightSideOfCharacterClassRange()
-        {
-            // Parsing the right hand side of a - is extremely strange (and most likely buggy) in
-            // the .NET parser. Specifically, the .NET parser will still consider itself on the
-            // right side no matter how many escaped dashes it sees.  So, for example, the following
-            // is legal [a-\-] (even though \- is less than 'a'). Similarly, the following are
-            // *illegal* [b-\-a] and [b-\-\-a].  That's because the range that is checked is
-            // actually "b-a", even though it has all the \- escapes in the middle.
-            //
-            // This is tracked with: https://github.com/dotnet/corefx/issues/31786
-
-            var first = ParseSingleCharacterClassComponent(isFirst: false, afterRangeMinus: true);
```

Comment:
this line moved to the single caller.  All teh remainder is what is no longer necessary as corefx no longer has the behavior of continually ignoring a sequence of `\-` constructs after the `-` in the range expression.